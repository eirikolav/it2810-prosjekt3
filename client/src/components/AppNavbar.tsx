import React, {Component, useState} from 'react'
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    Container
  } from 'reactstrap';

    export default function AppNavbar(props: any) {
        const [isOpen, setIsOpen] = useState(false)

        const toggle = () => {
           setIsOpen(!isOpen)
        }


        //expand="sm" is set to kick in for small screens
        //Adds mb-5, will move everything under the navbar
        return (
            <div>
            
            <Navbar color="dark" dark expand="sm" className="mb-5">
                <Container>
                    <NavbarBrand href="/">Shoppinglist</NavbarBrand>
                    <NavbarToggler onClick={toggle}/>
                    <Collapse isOpen={isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <NavLink href="https://github.com/ferdinwa">
                                    Github
                                </NavLink>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Container>
            </Navbar>
            
            </div>
        )
    }
    
        

  