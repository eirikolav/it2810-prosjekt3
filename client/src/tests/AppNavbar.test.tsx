import React from "react";
import renderer from "react-test-renderer";
import AppNavbar from "../components/AppNavbar";

test("Test AppNavbar", () => {
  const tree = renderer.create(<AppNavbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
