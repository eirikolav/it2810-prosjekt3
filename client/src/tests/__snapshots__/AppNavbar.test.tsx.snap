// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppNavbar 1`] = `
<div>
  <nav
    className="navbar navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        className="navbartext navbar-brand"
        href="/"
      >
        FH
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www.youtube.com/watch?v=ZJHUn8cV13w"
              onClick={[Function]}
            >
              <img
                alt=""
                className="img"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/f6/FIFA_21_logo.svg/1200px-FIFA_21_logo.svg.png"
              />
               
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
